external_server/server.py

```
@@ -0,0 +1,879 @@
+                )
+            self._get_and_handle_connect_message()
+            self._get_all_first_statuses_and_respond()
+            self._send_first_commands_and_get_responses()
+
+            self._set_state(ServerState.INITIALIZED)
+            if self._state == ServerState.INITIALIZED:
+                carlogger.info("Connect sequence has finished succesfully.", self._car)
+            self._event_queue.clear()
+        except Exception as e:
+            msg = f"Connection sequence has failed. {e}"
+            raise ConnectSequenceFailure(msg)
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)